acceptance-tests/proxy_protocol_test.go

```diff
@@ -61,9 +60,51 @@ var _ = Describe("Proxy Protocol", func() {
 		By("Sending a request without Proxy Protocol Header to HAProxy health check port")
 		_, err = http.Get(fmt.Sprintf("http://%s:8080/health", haproxyInfo.PublicIP))
 		expectConnectionResetErr(err)
+
 	})
 })
 
+var _ = Describe("Proxy Protocol with specified expect_proxy and accept_proxy=false", func() {
+	opsfileProxyProtocol := `---
+# Enable Proxy Protocol for specified CIDRs
+- type: replace
+  path: /instance_groups/name=haproxy/jobs/name=haproxy/properties/ha_proxy/accept_proxy?
+  value: false
+- type: replace
+  path: /instance_groups/name=haproxy/jobs/name=haproxy/properties/ha_proxy/expect_proxy?
+  value:
+    - 127.0.0.1/8
+    - ::1/128
+`
+	It("Correctly proxies Proxy Protocol requests", func() {
+		haproxyBackendPort := 12000
+		haproxyInfo, _ := deployHAProxy(baseManifestVars{
+			haproxyBackendPort:    haproxyBackendPort,
+			haproxyBackendServers: []string{"127.0.0.1"},
+			deploymentName:        deploymentNameForTestNode(),
+		}, []string{opsfileProxyProtocol}, map[string]interface{}{}, false)
+
+		closeLocalServer, localPort := startDefaultTestServer()
+		defer closeLocalServer()
+
+		closeTunnel := setupTunnelFromHaproxyToTestServer(haproxyInfo, haproxyBackendPort, localPort)
+		defer closeTunnel()
+
+		By("Waiting for monit to report that HAProxy is healthy")
+		// Since the backend is now listening, HAProxy healthcheck should start returning healthy
+		// and monit should in turn start reporting a healthy process
+		// We will wait up to one minute for the status to stabilise
+		Eventually(func() string {
+			return boshInstances(deploymentNameForTestNode())[0].ProcessState
+		}, time.Minute, time.Second).Should(Equal("running"))
+
+		By("Sending a request with Proxy Protocol Header to HAProxy traffic port")
+		err := performProxyProtocolRequest(haproxyInfo.PublicIP, 80, "/")
```

Comment:
in addition, we could have a test here that a regular (i.e. non-proxy-protocol) request fails, because HAProxy was in fact expecting proxy protocol.

Reply:
@peanball Thanks

Reply:
done